apps/condo/domains/miniapp/schema/B2BAccessToken.js

```diff
@@ -0,0 +1,258 @@
+    },
+    SERVICE_USER_DOES_NOT_MATCH_CONTEXT: {
+        code: BAD_USER_INPUT,
+        type: ACCESS_TOKEN_SERVICE_USER_DOES_NOT_MATCH_CONTEXT,
+        message: 'User and B2BAppContext connected to different B2BApps',
+    },
+    NEGATIVE_TTL: {
+        code: BAD_USER_INPUT,
+        type: ACCESS_TOKEN_NEGATIVE_TTL,
+        message: 'TTL can\'t be negative. You provided expiresAt datetime previous to the current one',
+    },
+}
+
+const ERRORS_FOR_DELETION = new Set([
+    ERRORS.CONTEXT_NOT_EXISTS,
+    ERRORS.RIGHT_SET_NOT_EXISTS,
+    ERRORS.NEGATIVE_TTL,
+])
+
+const B2BAccessToken = new GQLListSchema('B2BAccessToken', {
+    schemaDoc: 'Access to service user with additional restrictions by Organization and B2BAppAccessRightSet',
+    fields: {
+        sessionId: {
+            schemaDoc: 'Encrypted sessionId of session',
+            type: 'EncryptedText',
+            encryptionManager: encryptionManager,
+            isRequired: true,
+            access: userIsAdmin,
+            hooks: {
+                resolveInput ({ operation, existingItem, fieldPath }) {
+                    if (operation === 'create') {
+                        return ACCESS_TOKEN_SESSION_ID_PREFIX + crypto.randomUUID()
+                    }
+                    return get(existingItem, fieldPath)
+                },
+            },
+        },
+
+        token: {
+            schemaDoc: 'Token, ready to be put in Authorization header. Shown only once after creation',
+            type: 'Virtual',
+            graphQLReturnType: 'String',
+            resolver: (item) => {
+                const sessionId = encryptionManager.decrypt(item.sessionId)
+                return cookieSignature.sign(sessionId, COOKIE_SECRET)
+            },
+            access: access.readableOnlyOnCreate,
+        },
+
+        context: {
+            schemaDoc: 'Link to B2BApp and Organization',
+            type: 'Relationship',
+            ref: 'B2BAppContext',
+            isRequired: true,
+            knexOptions: { isNotNullable: true }, // Required relationship only!
+            kmigratorOptions: { null: false, on_delete: 'models.CASCADE' },
+            access: access.updatableOnlyForAdmin,
+        },
+
+        user: {
+            ...SERVICE_USER_FIELD,
+            isRequired: true,
+            access: access.updatableOnlyForAdmin,
+            hooks: {
+                ...SERVICE_USER_FIELD.hooks,
+                async resolveInput ({ operation, existingItem, resolvedData, fieldPath }) {
+                    const newItem = {
+                        ...existingItem,
+                        ...resolvedData,
+                    }
+                    if (operation === 'create' && !newItem[fieldPath]) {
+                        const contextId = get(resolvedData, 'context') || get(existingItem, 'context')
+                        const context = await getById('B2BAppContext', contextId)
```

Comment:
Add test for that

Reply:
I have no idea how to write this kind of test without random fails.
From my understanding getById returning deleted entity should be super rare case, because it is being validated in `validateInput`

Reply:
I don't want a "super rare case" to even exist, I want logic to work 100% of time, so once you implement it good, it can be left for years